nemo_skills/pipeline/utils/declarative.py

```
@@ -492,6 +616,321 @@ def run(self, dry_run: bool = False, log_dir: Optional[str] = None, _reuse_exp=N
+            handle = backend.submit_job(k8s_job_spec)
+            # Track by ORIGINAL name for dependency lookups
+            job_name_to_handle[original_job_name] = handle
+            LOG.info(f"Submitted job '{original_job_name}' as '{job_name}' (job_id: {handle.job_id})")
+
+            # If sequential, wait for this job to complete before continuing
+            if sequential:
+                LOG.info(f"Waiting for job '{job_name}' to complete (sequential mode)...")
+                status = backend.wait_for_completion(handle)
+                LOG.info(f"Job '{job_name}' completed with status: {status.value}")
+                if status == JobStatus.FAILED:
+                    raise RuntimeError(f"Job '{job_name}' failed, aborting pipeline")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)